### PR TITLE
Fix `RadioCard` styles and interactive states

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

This PR fixes the tokens applied to the `RadioCard` in it's selected state and updates the color of text elements within a disabled `RadioCard`.

### :hammer_and_wrench: Detailed description

Previously the `RadioCard` was using `focus-action-internal` for the `border-color` when a `RadioCard` was selected. Not only is this the incorrect token, but it differs from Figma and the interactive states of the component. This change updates the border color for a selected `RadioCard` to:

- `blue-300` for resting/default state
- `blue-400` for hover state

Additionally, we are proposing in design to change the tokens for text elements (namely the icon, `description`, and `label` within the `RadioCard`) to use `form-control-disabled-foreground-color` to semantically match the rest of the tokens applied when the component is disabled.

### :camera_flash: Screenshots

**Before**
Screenshot 2024-09-23 at 1 24 32 PM

**After**
Screenshot 2024-09-23 at 1 24 06 PM

### :link: External links

Jira ticket: [HDS-3868](https://hashicorp.atlassian.net/browse/HDS-3868)
Figma file: [Update RadioCard disabled colors](https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/3ctSmHIlVaRYqa8E6OaIaG/HDS-Components?m=auto&node-id=16838-51020&t=JkKW71GXhXPqXkeX-1)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3868]: https://hashicorp.atlassian.net/browse/HDS-3868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/styles/components/form/radio-card.scss [hunk 1]
```diff
@@ -64,21 +64,32 @@
 
   &--checked,
   &.mock-checked {
-    border-color: var(--token-color-focus-action-internal);
+    border-color: var(--token-color-palette-blue-300);
```
Comment (jorytindall) [inline packages/components/src/styles/components/form/radio-card.scss:67]: Update the default border color for a selected `RadioCard`

File: packages/components/src/styles/components/form/radio-card.scss [hunk 2]
```diff
@@ -64,21 +64,32 @@
...
 
     .hds-form-radio-card__control-wrapper {
       background-color: var(--token-color-surface-action);
       border-color: var(--token-color-border-action);
     }
+
+    &:hover,
+    &.mock-hover {
+      border-color: var(--token-color-palette-blue-400);
```
Comment (jorytindall) [inline packages/components/src/styles/components/form/radio-card.scss:76]: Add the interactive style on hover.

File: packages/components/src/styles/components/form/radio-card.scss [hunk 3]
```diff
@@ -64,21 +64,32 @@
...
+    }
   }
 
   &--disabled,
   &.mock-disabled {
     background-color: var(--token-color-surface-interactive-disabled);
     border-color: var(--token-color-border-primary);
     box-shadow: none;
+    color: var(--token-form-control-disabled-foreground-color);
```
Comment (jorytindall) [inline packages/components/src/styles/components/form/radio-card.scss:85]: Set the base color for child elements within the `RadioCard` in it's disabled state.

File: packages/components/src/styles/components/form/radio-card.scss [hunk 4]
```diff
@@ -64,21 +64,32 @@
...
     cursor: not-allowed;
 
+    .hds-form-radio-card__label,
+    .hds-form-radio-card__description {
+      color: var(--token-form-control-disabled-foreground-color);
```
Comment (jorytindall) [inline packages/components/src/styles/components/form/radio-card.scss:90]: Overwrite with specificity the `label` and `description` text elements.

**Review by heatherlarsen — APPROVED** [review]
lgtm 👍🏻

**Review by alex-ju — APPROVED** [review]
Code changes look good!

One note, as you mentioned, the disabled overrides are only applied to labels and descriptions; a coloured icon or badge (or any custom element for that matter) would still have the same aspect.

Screenshot 2024-09-24 at 15 14 45

**Comment by jorytindall** [comment]
@alex-ju 
> One note, as you mentioned, the disabled overrides are only applied to labels and descriptions; a coloured icon or badge (or any custom element for that matter) would still have the same aspect.

Yes, I did think about this and there didn't really seem like a systematic way to approach that so I kept the scope specific to those elements.

**Comment by didoo** [comment]
> @alex-ju
> 
> > One note, as you mentioned, the disabled overrides are only applied to labels and descriptions; a coloured icon or badge (or any custom element for that matter) would still have the same aspect.
> 
> Yes, I did think about this and there didn't really seem like a systematic way to approach that so I kept the scope specific to those elements.

Don't have full context (and I may have mentioned elsewhere, so forgive me if that's the case) but one trick I used to "emulate" a disabled state of something was to change its opacity to 50/40%, so it would blend in with the background; maybe something that can be done to the icon? worth a try?

**Comment by jorytindall** [comment]
@didoo interesting suggestion, I played around with this and while I don't think it's feasible to target every nested component (especially when generic content is passed), it seems pragmatic to target the icon and the badge directly with this method. I updated the example in the showcase for reference, let me know what you think of the method and I'll revert it before committing (unless you see a benefit in including an example like this).

**Comment by didoo** [comment]
> @didoo interesting suggestion, I played around with this and while I don't think it's feasible to target every nested component (especially when generic content is passed), it seems pragmatic to target the icon and the badge directly with this method. I updated the example in the showcase for reference, let me know what you think of the method and I'll revert it before committing (unless you see a benefit in including an example like this).

@jorytindall (+ @alex-ju) you may consider changing the opacity of the whole `.hds-form-radio-card__content` container, instead of targeting its children one by one (which means leaving the text foreground color as “strong/primary”); this would result in something like this:
screenshot_4220

**Comment by alex-ju** [comment]
if acceptable from a design standpoint, changing the opacity for the entire card seems like a more robust approach

**Pull request merged**